apis/v1alpha2/httproute_types.go

```diff
@@ -461,14 +466,20 @@ type HTTPRouteMatch struct {
 	// ANDed together, meaning, a request must match all the specified headers
 	// to select the route.
 	//
+	// +listType=map
+	// +listMapKey=name
 	// +optional
+	// +kubebuilder:validation:MaxItems=16
 	Headers []HTTPHeaderMatch `json:"headers,omitempty"`
```

Comment:
That is a lot of parameters to base a match on practically. No need to change, I don't feel strongly enough to ask for a change.

Reply:
Yeah I almost started with something lower, but didn't have a great reason. We use 16 for a lot of similar limits so it seemed like a relatively non-controversial starting point.